src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -1,46 +1,88 @@
+  case class ProvideCallAbortButton(comboID: ComboID, cancelButton: ()=>Unit) extends CallActorMessage
+  case class AbortCall(workflowId: WorkflowID)
+  case class ExecutionFinished(callExecutionId: UUID, call: Call, outputs: Try[CallOutputs]) extends CallActorMessage
 
-  def props(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor): Props =
-    Props(new CallActor(call, locallyQualifiedInputs, backend, workflowDescriptor))
+  def props(): Props = Props(new CallActor())
 }
 
 /** Actor to manage the execution of a single call. */
-class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) extends Actor with CromwellActor {
+class CallActor() extends Actor with CromwellActor {
+
+  import CallActor._
 
   type CallOutputs = Map[String, WdlValue]
+  type CancelButton = ()=>Unit
+  val callActorId: UUID = UUID.randomUUID()
+
+  // NB: Should only have one entry but the list is useful for safe mutability.
+  var cancelButtonList: mutable.MutableList[(ComboID, CancelButton)] = new mutable.MutableList[(ComboID, CancelButton)]()
 
   private val log = Logging(context.system, classOf[CallActor])
-  val tag = s"CallActor [UUID(${workflowDescriptor.shortId}):${call.name}]"
+  val tag = s"CallActor [UUID($callActorId)]"
 
   override def receive = LoggingReceive {
-    case CallActor.Start => handleStart()
-    case CallActor.ExecutionFinished(outputs) => context.parent ! WorkflowActor.CallCompleted(call, outputs)
-    case CallActor.ExecutionFailed(regret) =>
-      log.error(regret, s"$tag: ${regret.getMessage}")
-      context.parent ! WorkflowActor.CallFailed(call, regret.getMessage)
+    case CallActor.Start(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) =>
+      handleStart(call, locallyQualifiedInputs, backend, workflowDescriptor)
+    case ProvideCallAbortButton(comboId: ComboID, cancelButton: CancelButton) => handleAbortButton(comboId, cancelButton)
+    case CallActor.ExecutionFinished(callExecutionId, call: Call, outputTry) => handleFinished(callExecutionId, call, outputTry)
+    case CallActor.AbortCall(workflowId: WorkflowID) => handleAbort(workflowId)
     case badMessage =>
       val diagnostic = s"$tag: unexpected message $badMessage."
       log.error(diagnostic)
-      context.parent ! engine.workflow.WorkflowActor.CallFailed(call, diagnostic)
+  }
+
+  private def handleFinished(callExecutionId: CallExecutionID, call: Call, outputTry: Try[CallOutputs]) = {
+    cancelButtonList = cancelButtonList filterNot { _._1._1.equals(callExecutionId) }
+
+    outputTry match {
+      case Success(outputs) => context.parent ! WorkflowActor.CallCompleted(call, outputs)
+      case Failure(regret) =>
```

Comment:
I think this might look a little cleaner if you do the interior match in the parent, e.g.:

    case Failure(e: TaskAbortedException) => ???
    case Failure(e: Throwable) => ???